### ISSUE TITLE
When setting the color depth of X to 30 (10-bit), panel is not clickable.

### ISSUE THREAD

**Issue by lss4** [linuxmint/muffin#534]
```
 * Cinnamon version (cinnamon --version)
4.0.9 (at the time of posting)
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Manjaro (latest)
 * Graphics hardware *and* driver used
nvidia / amdgpu
(The issue has been observed on two machines, both with latest Manjaro Cinnamon, one with amdgpu, the other with nvidia, so it's probably irrelevant to graphic cards)
 * 32 or 64 bit
64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
(I don't think there's anything "erroneous" to be recorded here. However, I'm fine with providing logs when needed, if I can be instructed on where to get them)
 ```

**Issue**
Originally posted on [Manjaro forum](https://forum.manjaro.org/t/cinnamon-panel-not-clickable-when-setting-monitor-to-enable-10-bit-display/77511), but someone suggested me to post it here as it looked more like a bug with Cinnamon.

When booting with X configured to use a color depth of 30 (as opposed to 24) the panel (Taskbar/Menu) is not clickable. However, I can access a limited amount of menu contents using keyboard, such as WIN key, arrow and Enter keys. Plus, during this scenario, clicking on the menu that I opened using WIN key would instead close it.

By the way, even with a 30-bit desktop, I can click the icons on the desktop (such as Computer) without issues, as well as perform clicking inside file manager and other apps. The issues I’m having seem to match those mentioned [here](https://devtalk.nvidia.com/default/topic/1023145/linux/editing-xorg-conf-defaultdepth-and-depth-to-30-makes-the-mouse-buttons-stop-working/), although it's mostly GNOME-related.

**Steps to reproduce**

1. Set X to use 10-bit (Depth 30) following [this example](https://bbs.archlinux.org/viewtopic.php?id=241007) (on Arch forum). Save the necessary settings into a custom conf file in "/etc/X11/xorg.conf.d/".
2. Reboot the system
3. Confirm 10-bit status: Through xwininfo I can confirm the system is indeed operating on 10-bit (test videos generated using [this repo](https://github.com/jursonovicst/gradient/) also confirmed working 10-bit).
4. When booting with X color depth set to 10-bit (30), the panel is not clickable anymore. However, I can still click on desktop icons as well as in file manager and other programs without issues, implying the issue only affected the panel

**Expected behaviour**

The panel should be properly clickable just like the rest of the system.

**Other information**

I don't have this issue on Manjaro XFCE and amdgpu. XFCE's panel works without issues. Not sure how many people actually used 10-bit, but from the looks of it, 10-bit is rather recent and there are programs that have known issues with it (some have workarounds).

**Comment by mabod** [comment]
I do not believe that this is an xorg or gnome issue. None of the issues you have linked apply to me.

xorg is working fine without any noticeable mouse problem. gnome and mutter are working fine without any noticeable problem. 
xorg-server 1.20.5-2.1; 
gnome-shell 1:3.32.2+11+g1c6abf378-1; 
mutter 3.32.2+43+gb7f158811-1

It is just the cinnamon panel that does not react on mouse clicks. While a right mouse click on the background is working. And if I open a terminal via the right mouse click popup meu, I can handle the terminal window with the mouse just fine. 

And by the way, with 30 bit color depth the background image is not displayed. The background is simply black.

**Comment by leigh123linux** [comment]
@mabod If your sure I wont waste time trying to backport the mutter commits to muffin.

https://bugzilla.gnome.org/show_bug.cgi?id=743620

**Comment by mabod** [comment]
What can I say: I am working in gnome the whole evening with 30 bit (according to xwininfo) and I did not find any issue with mouse or anything.

**Comment by TheTrustedComputer** [comment]
I have the same issue as described by the original poster, except that the wallpaper is fully functional. I am running Cinnamon 5.0.7 in Manjaro with an AMD Radeon RX 5500 XT on two HDR10 capable monitors. This graphics card supports 30-bit color depth, yet the panel is unresponsive to mouse inputs when running under 30-bit color.

**Comment by TeaCult** [comment]
Same here desktop wallpaper gone , panel is not clickable , moved and backed up .cinnamon nothing is changed.

**Comment by lss4** [comment]
```
$ cinnamon --version
Cinnamon 6.6.7
```

Sorry for not checking about this issue. It's been years since I reported it, and it's been years since I actively used 10-bit on Linux as back then there were just too many software that did not like it. As such, I kind of forgot about this issue until I was notified today that it's been automatically closed for inactivity.

**TL;DR:** The issue is apparently fixed with current Cinnamon versions. Panel is correctly clickable in 10-bit mode. No apparent issues.

I just did a test enabling 10-bit on my system via a custom `xorg.conf.d` config file. Can confirm it's indeed using 10-bit.

```
$ cat /var/log/Xorg.0.log | grep AMDGPU
[    14.326] (II) AMDGPU: Driver for AMD Radeon:
[    14.370] (**) AMDGPU(0): Depth 30, (--) framebuffer bpp 32
[    14.370] (II) AMDGPU(0): Pixel depth = 30 bits stored in 4 bytes (32 bpp pixmaps)
[    14.370] (==) AMDGPU(0): Default visual is TrueColor
[    14.370] (==) AMDGPU(0): RGB weight 101010
[    14.370] (II) AMDGPU(0): Using 10 bits per RGB (8 bit DAC)
[    14.370] (--) AMDGPU(0): Chipset: "AMD Radeon RX 5700 XT 50th Anniversary" (ChipID = 0x731f)
[    14.600] (II) AMDGPU(0): glamor: Using OpenGL 4.6 context.
[    14.600] (II) AMDGPU(0): glamor X acceleration enabled on AMD Radeon RX 5700 XT 50th Anniversary (radeonsi, navi10, ACO, DRM 3.64, 6.19.11-arch1-1-znver3)
[    14.600] (II) AMDGPU(0): glamor detected, initialising EGL layer.
[    14.600] (==) AMDGPU(0): TearFree property default: auto
[    14.600] (==) AMDGPU(0): VariableRefresh: disabled
[    14.600] (==) AMDGPU(0): AsyncFlipSecondaries: disabled
[    14.600] (II) AMDGPU(0): KMS Pageflipping: enabled
[    14.604] (II) AMDGPU(0): Output DisplayPort-0 using monitor section Monitor0
[    14.604] (II) AMDGPU(0): Output DisplayPort-1 has no monitor section
[    14.604] (II) AMDGPU(0): Output DisplayPort-2 has no monitor section
[    14.604] (II) AMDGPU(0): Output HDMI-A-0 has no monitor section
[    14.640] (II) AMDGPU(0): EDID for output DisplayPort-0
[    14.640] (II) AMDGPU(0): Manufacturer: AUS  Model: 27ba  Serial#: 16843009
[    14.640] (II) AMDGPU(0): Year: 2021  Week: 22
[    14.640] (II) AMDGPU(0): EDID Version: 1.4
[    14.640] (II) AMDGPU(0): Digital Display Input
[    14.640] (II) AMDGPU(0): 10 bits per channel
[    14.640] (II) AMDGPU(0): Digital interface is DisplayPort
[    14.640] (II) AMDGPU(0): Max Image Size [cm]: horiz.: 60  vert.: 34
[    14.640] (II) AMDGPU(0): Gamma: 2.20
[    14.640] (II) AMDGPU(0): DPMS capabilities: Off
[    14.640] (II) AMDGPU(0): Supported color encodings: RGB 4:4:4 YCrCb 4:4:4 YCrCb 4:2:2
[    14.640] (II) AMDGPU(0): First detailed timing is preferred mode
[    14.640] (II) AMDGPU(0): Preferred mode is native pixel format and refresh rate
[    14.640] (II) AMDGPU(0): Display is continuous-frequency
[    14.640] (II) AMDGPU(0): redX: 0.688 redY: 0.297   greenX: 0.271 greenY: 0.659
[    14.640] (II) AMDGPU(0): blueX: 0.148 blueY: 0.040   whiteX: 0.312 whiteY: 0.328
...
```

This is on my current system, using Garuda Cinnamon. After starting up I checked the panel and it's working correctly. I can properly click on stuffs on the panel and so far haven't noticed anything out of ordinary. So this can be considered fixed for now.

On the other hand, 10-bit compatibility with software has indeed been improved. Chromium is working correctly. Office suites like LibreOffice and OnlyOffice have no apparent glitches from what I can see for now. Also, it seems VSCode is working okay without apparent issues.

Some issues I noticed at the moment with 10-bit mode:
- Colors in KiCAD (Schematic and PCB Editor) are messed up and not really in a usable condition (everything gets a bright red tint making it very difficult to read or work on). Can still use it for other purposes...
- Java software that uses `sun.java2d` still refuse to work out-of-box throwing that good old `InvalidPipeException` with message `Unsupported bit depth: 30`, with JDK21 in year 2026.

Anyway... I'm thinking of reverting to default once I've done testing, as right now I still have need for KiCAD.

EDIT: It seems KiCAD Classic theme works for Schematic Editor to some extent, giving a proper white background although text and other stuffs are still in red. I can't find any good color scheme for the PCB/Footprint Editor (as well as Gerber Viewer), however.
